Sources/Config/ConfigurationProperty.swift

```diff
@@ -26,21 +29,35 @@ struct ConfigurationProperty<T>: Property, AssociatedPropertyKeyProviding {
         }
     }
 
+    private static func transformValueToType(value: Any?) throws -> T {
+        if let val = value as? T {
+            return val
+        }
+        if let nilLiteralType = T.self as? ExpressibleByNilLiteral.Type {
```

Comment:
Can you add some doc comments on this. I've read it about 10 times and think I'm understnading it now, but doesn't hurt to add something in to explain what is a pretty opaque piece of code without significant context.

Reply:
In the end, I'm assuming it's entirely because you can't convert `NSNull` to `nil`?

Reply:
Yeah, this was fun...

This is to handle the case where `T: Optional`. If the typecast to `T` fails (in this case `NSNull.Type` -> `Optional<String>`) we want to return `nil`, but in a way that isn't a failure. 

`nil` can't be assigned to `defaultValue` for example, because as far as `ConfigurationProperty` knows it's not optional. We also can't construct the `Optional` directly, because we don't know the `Wrapped` type.

Reply:
Hopefully a bit clearer with that documentation.